### PR TITLE
Fix a TODO about using temlateId type arguments

### PR DESCRIPTION
**Pull request by hurryabit**
Also fix an unrelated tiny outdated instruction in the README.

---
This change is [Reviewable](https://reviewable.io/reviews/digital-asset/davl/234)

### CODE REVIEW

File: ui/src/utils/vacation.ts [hunk 1]
```diff
@@ -15,17 +15,25 @@ export type Vacation = {
...
-// the next release.
 export type VacationCreateEvent =
-  | CreateEvent<v4.DAVL.Vacation, undefined>
-  | CreateEvent<v5.DAVL.V5.Vacation, undefined>;
+  | CreateEvent<v4.DAVL.Vacation, undefined, typeof v4.DAVL.Vacation.templateId>
+  | CreateEvent<
+      v5.DAVL.V5.Vacation,
+      undefined,
+      typeof v5.DAVL.V5.Vacation.templateId
+    >;
```
Comment (rohanjr) [inline ui/src/utils/vacation.ts]: Nit: Use the same formatting for both alternatives? Or is this the result of auto formatting?

File: ui/src/utils/vacation.ts [hunk 2]
```diff
@@ -38,10 +46,6 @@ export const makeVacation = <T extends object>(
     case v5.DAVL.V5.Vacation.templateId:
       version = "v5";
       break;
-    default:
-      // NOTE(MH): This is the fallout of the above comment on the third
-      // paramter of `CreateEvent`.
-      throw Error(`Invalid template id for Vacation: ${contract.templateId}`);
```
Comment (rohanjr) [inline ui/src/utils/vacation.ts]: Does TypeScript check that the cases are exhaustive and allow you to remove the default case now?

File: ui/src/utils/vacation.ts (hunk 1 above)
Comment (hurryabit) [reply ui/src/utils/vacation.ts]: It is indeed the result of auto-formatting.

File: ui/src/utils/vacation.ts (hunk 2 above)
Comment (hurryabit) [reply ui/src/utils/vacation.ts]: Yes, it does. Isn't that awesome? That's why I made the change.

Comment (rohanjr) [reply ui/src/utils/vacation.ts]: 😎

**Pull request merged**